fs_overlay/bin/archname

```diff
@@ -0,0 +1,80 @@
+#/bin/sh
+# Usage: archname.sh <PACKAGE>
+#
+# Outputs the correct architecture string for the named package, translating
+# between the different platform strings. The nice things about standards…
+# PACKAGE is one of "s6-overlay" or "docker-gen"
+#
+# Reads the docker architecture from $TARGETPLATFORM (set by `docker buildx
+# build`) and outputs the appropriate per-package architecture name. If
+# $TARGETPLATFORM is unset (using legacy `docker build`), it is determined from
+# the output for `arch`, which is unreliable in case it is run in an emulator.
+# If you really plan to build inside an emulator with the legacy build process,
+# you are on your own.
+
+
+# Platform compatibility/naming matrix as of 2020-01-17:
+# - Nginx: https://hub.docker.com/_/nginx?tab=tags
+# - S6 Overlay: https://github.com/just-containers/s6-overlay/releases
+# - Docker-Gen: https://github.com/jwilder/docker-gen/releases
+# - ACME-Tiny: No architecture/platform dependencies
+# - Stars elsewhere are not part of the output, but indicate special values
+#
+# Nginx (Docker) | `arch` output | S6 Overlay | Docker-Gen | ACME-Tiny
+# ---------------+---------------+------------+------------+----------
+# linux/386      | x86_64*       | x86        | i386       | *
```

Comment:
What does the `*` mean here?

When `Nginx` image is `linux/386`, `arch` command outputs `x86_64`. Does it mean Nginx Docker image put `linx/386` Nginx binary in a x86_64 container?

Reply:
I have never used the Dockerhub build process. I use `docker build buildx --push …` (performed by  `make docker-multiarch`) to build and publish the multi-platform image it to Dockerhub.

Reply:
Regarding `arch` output, the notes below the table say:
```
# - `arch` output as reported potentially inside the emulater with
#   `docker buildx build --builder docker-multiarch --platform ...` 
```
`arch` in QEMU containers may output a backward-compatible platform instead of the actual platform requested.
So, `docker build` will work correctly (`archname` uses `arch`, which is reliable in these cases) when not cross-building (has to rely on `arch` output); anyone interested in (foreign/multi)platform builds has to use `docker buildx` (and probably will do that anyway, as `archname` has to rely on `TARGETPLATFORM`, as `arch` might output a more powerful platform).

Otherwise said, on the real hardware, `arch` is expected to output the correct value, but not in an emulation. And when using `docker build` on real hardware, the right image is created. And when using `docker buildx`, the right image is created, independent of the underlying real hardware.